assets/js/script.js

```diff
@@ -15,19 +16,61 @@ var formHandler = function (event) {
         console.log("sent to movies");
         // send searchTerm to Mason's movie fetch function
         // movieFetchHandler(searchTerm);
-    }
-    else if (selectedMedia === "music") {
+    } else if (selectedMedia === "music") {
         console.log("sent to music");
         // send searchTerm to music fetch function
         // musicFetchHandler(searchTerm);
-    }
-    else if (selectedMedia === "books") {
+    } else if (selectedMedia === "books") {
         console.log("sent to books");
         // send searchTerm to book fetch function
         bookFetchHandler(searchTerm);
     }
 };
 
+// function to fetch book data using user input as parameter
+var bookFetchHandler = function (searchTerm) {
+    if (searchByEl.value === "title") {
+        var apiUrl = "https://www.googleapis.com/books/v1/volumes?q=" +
+            searchTerm;
+        console.log(apiUrl);
+        fetch(apiUrl)
+            .then(function (response) {
+                // request was successful
+                if (response.ok) {
+                    response.json().then(function (data) {
+                        // send data to function which will create object of
+                        // relevent information
+                        // bookObjectCreator(data);
+                    });
+                } else {
+                    alert("Error: " + response.statusText);
+                }
+            })
+            .catch(function (error) {
+                alert("Unable to connect");
+            });
+    } else if (searchByEl.value === "author") {
+        var apiUrl = "https://www.googleapis.com/books/v1/volumes?q=" +
+            searchTerm +
+            "+inauthor:" + searchTerm;
+        console.log(apiUrl);
+        fetch(apiUrl)
+            .then(function (response) {
+                // request was successful
+                if (response.ok) {
+                    response.json().then(function (data) {
+                        // send data to function which will create object of
+                        // relevent information
+                        // bookObjectCreator(data);
+                    });
+                } else {
+                    alert("Error: " + response.statusText);
+                }
+            })
+            .catch(function (error) {
+                alert("Unable to connect");
+            });
```

Comment:
You can reduce the amount of code by reusing the fetch function. The only thing needing to be in the conditional is the definition of the apiUrl.

Reply:
I tried that originally but because the apiUrl variable is being defined within the conditional, the call outside of the conditional couldn't reach it. If you can make it work I'd prefer that way also. Any ideas?

Reply:
intitialize the variable outside (and above) the conditional and define it inside the conditional. Example:
let apiUrl;
if {
apiUrl = ""
} else {
apiUrl = ""
}

Reply:
`let` or `var`

Reply:
Awesome I'll do that then let you know when I've pushed.